mhs/inbound/inbound/request/handler.py

```diff
@@ -136,16 +194,29 @@ def _extract_incoming_async_request_message(self):
 
         return request_message
 
-    def _extract_incoming_sync_request_message(self):
-        try:
-            request_message = SoapEnvelope.from_string(self.request.headers,
-                                                       self.request.body.decode())
-        except SoapParsingError as e:
-            logger.error('021', 'Failed to parse response: {exception}', {'exception': e})
-            raise tornado.web.HTTPError(500, 'Error occurred during message parsing',
-                                        reason=f'Exception during inbound message parsing {e}') from e
+    def _is_message_intended_for_receiving_mhs(self, request_message):
+        """
+        Asserts whether the incoming message was intended for thi MHS instance given the to party key defined in the
+        message.
 
-        return request_message
+        error_code, severity and description defined as per the TMS Error Base v3.0 document
 
-    def _is_async_request(self):
-        return 'multipart/related' in self.request.headers[CONTENT_TYPE_HEADER_NAME]
+        :param request_message: the parsed ebxml request message
+        """
+        return self.party_id == request_message.message_dictionary[ebxml_envelope.TO_PARTY_ID]
+
+    def _return_message_to_message_initiator(self, request_message):
+
+        if self.party_id != request_message.message_dictionary[ebxml_envelope.TO_PARTY_ID]:
+            try:
+                nack_context = {
+                    ebxml_envelope.ERROR_CODE: "ValueNotRecognized",
+                    ebxml_envelope.DESCRIPTION: "501314:Invalid To Party Type attribute",
```

Comment:
@colin-bateson document where these come from.

Also, this description is already hardcoded in the template

Reply:
This was already in the doc string but also added inline. 

It isn't hardcoded in the mustache template... only in the test ones.